mux_test.go

```
@@ -0,0 +1,430 @@
+			}
+			return true
+		})
+		WorkerRegistry.Clear()
+	}
+}
+
+func TestMuxHealthCheck(t *testing.T) {
+	server, cleanup := setupMuxServer()
+	defer cleanup()
+
+	req, err := http.NewRequest("GET", "/health", nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)